crates/component/src/wrappers/http.rs

```diff
@@ -0,0 +1,422 @@
+//! This module provides utilities for writing HTTP servers and clients using the WASI HTTP API.
+//!
+//! It's inspired by the WASI 0.3 proposal for <https://github.com/WebAssembly/wasi-http> and will
+//! be supported until the release of wasi:http@0.3.0. After that, this module will be deprecated.
+//!
+//! This crate can be used with the [reqwest-wasmcloud](https://github.com/wasmCloud/reqwest-wasmcloud)
+//! crate to make outgoing HTTP requests from a WebAssembly component.
+//!
+//! ```rust
+//! use wasi::http::types::ErrorCode;
+//! use wasmcloud_component::http::{HttpServer, Request, ResponseBuilder};
+//!
+//! struct Component;
+//!
+//! wasmcloud_component::http::export!(Component);
+//!
+//! // Implementing the [`HttpServer`] trait for a component
+//! impl HttpServer for Component {
+//!     fn handle(_request: Request) -> Result<ResponseBuilder, ErrorCode> {
+//!         Ok(ResponseBuilder::ok("Hello from Rust!"))
+//!     }
+//! }
+//! ```
+// Required until <https://github.com/seanmonstar/reqwest/pull/2453> merges
+use reqwest_wasmcloud as reqwest;
+
+use std::{
+    io::{Read, Write},
+    str::FromStr,
+};
+
+use reqwest::header::{HeaderMap, HeaderName, HeaderValue};
+use wasi::{
+    http::{
+        outgoing_handler::ErrorCode,
+        types::{
+            Fields, IncomingBody, InputStream, Method, OutgoingBody, OutgoingResponse,
+            ResponseOutparam, Scheme,
+        },
+    },
+    io::streams::StreamError,
+};
+
+const HTTP_SCHEME: &str = "http";
```

Comment:
Why not use https://docs.rs/http/latest/http/uri/struct.Scheme.html#associatedconstant.HTTP instead?